src/utils/context.ts

```diff
@@ -0,0 +1,247 @@
+ * @param node
+ * @return true if a favorite context root, false otherwise
+ */
+export function isFavoriteContext(node: TreeItem): boolean {
+    return new RegExp(extension.FAVORITE_CONTEXT).test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a dataset member
+ * @param node
+ * @return true if a dataset member, false otherwise
+ */
+export function isDsMember(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_MEMBER_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a dataset session
+ * @param node
+ * @return true if a dataset session, false otherwise
+ */
+export function isDsSession(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_SESSION_CONTEXT + ")").test(node.contextValue);
+}
+
+
+/**
+ * Helper function which identifies if the node is a partitioned dataset
+ * @param node
+ * @return true if a partitioned dataset, false otherwise
+ */
+export function isPds(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_PDS_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a USS Directory
+ * @param node
+ * @return true if a USS Directory, false otherwise
+ */
+export function isUssDirectory(node: TreeItem): boolean {
+    return new RegExp("^" + extension.USS_DIR_CONTEXT).test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a USS session
+ * @param node
+ * @return true if a USS session, false otherwise
+ */
+export function isUssSession(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.USS_SESSION_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a grouping or folder
+ * @param node
+ * @return true if a folder, false otherwise
+ */
+export function isFolder(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.JOBS_JOB_CONTEXT + "|" + extension.USS_DIR_CONTEXT + "|"
+                        + extension.DS_PDS_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a session
+ * @param node
+ * @return true if a session, false otherwise
+ */
+export function isSession(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.JOBS_SESSION_CONTEXT + "|" + extension.USS_SESSION_CONTEXT + "|"
+                        + extension.DS_SESSION_CONTEXT + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a session but not a favorite
+ * @param node
+ * @return true if a session, false otherwise
+ */
+export function isSessionNotFav(node: TreeItem): boolean {
+    return new RegExp("^((?!.*" + extension.FAV_SUFFIX + ")(" + extension.JOBS_SESSION_CONTEXT + "|" + extension.USS_SESSION_CONTEXT + "|"
+                        + extension.DS_SESSION_CONTEXT + "))").test(node.contextValue);
+}
```

Comment:
A lot of these functions could reuse the other context check functions For example, could this function be
```js
export function isSessionNotFav(node: TreeItem): boolean {
    return isSession(node) && !isFavorite(node);
}
```

Reply:
They could, yes but I am concerned about performance so have tried to combine regular expression tests so they only run once. It may look more complex but is far more efficient

Reply:
Ah I didnt think about that.
I ran some simulations to test this and found these results. (Times are for all iterations)
```
iterations: 1,000,000
- 1 single regex: 244.296ms
- 2 regex with boolean logic: 376.171ms
```
So you are right doing one larger regex is faster. So we dont need to add this change if performance is a high priority.
I only found a difference of ~130ms over 1 million regex tests so its not a large difference